### ISSUE TITLE
custom type serializer for `numeric` is never called

### ISSUE THREAD

**Issue by rafaelcr** [porsager/postgres#341]
Following the readme [here](https://github.com/porsager/postgres#numbers-bigint-numeric), I'm trying to override how `bigint` values are inserted into `numeric` columns so I can insert them using strings (I was getting "out of range" errors for large numbers).

However, I'm not sure why the `serialize` function is never called in the following code:
```typescript
numeric: {
  to: 1700,
  from: [1700],
  serialize: (x: any) => typeof x === 'bigint' ? x.toString() : x,
  parse: (x: any) => BigInt(x),
},
```
The `parse` function works correctly (so do serializers for other types). Is there something I'm missing? I've also tried with other type `oid`s that might be related but no luck.

Thanks for the help on this and all other issues!

**Comment by porsager** [comment]
Hi Rafael. 

Could you post some code reproing the issue? I feel like I'm missing some context to understand what's wrong? 

It also seems the serialize in your example is unnecessary since, toString will be called anyway. 

Also, are you sure it's safe for you to cast numeric to bigint like that in parse?

**Comment by rafaelcr** [comment]
> Could you post some code reproing the issue?

@porsager sure, here's an example where I get this error returned: `PostgresError: value "9970000000000000000" is out of range for type bigint`:
```typescript
const values = {
  amount: BigInt(9970000000000000000)
}
await sql`INSERT INTO events ${sql(values)}`; // table `events` has an `amount` column of type `numeric`
```

However, if I do this, it works:
```typescript
const values = {
  amount: BigInt(9970000000000000000).toString()
}
await sql`INSERT INTO events ${sql(values)}`;
```

> Also, are you sure it's safe for you to cast numeric to bigint like that in parse?

You're right, I just wrote that for a quick example but I'll pass it as it is and let application logic decide whether or not to parse as `BigInt`

**Comment by porsager** [comment]
Ah, well that is outside the allowed Postgres `bigint` (8 bytes) range which is -9223372036854775808 to +9223372036854775807.

The JS BigInt type doesn't have a limit per se, so it doesn't map to postgres 1 to 1, so if you need to represent numbers outside that range you will probably need to use `numeric` instead.

There's not much I can do about this in Postgres.js I think.

**Comment by rafaelcr** [comment]
Gotcha, yeah, I assume postgres is trying to fit the number into a `bigint` even though the column is already `numeric`. I'll send them as strings as a workaround.

Thanks again

**Comment by porsager** [comment]
Ahh.. yes if you cast it to bigint somewhere along the way postgres will complain. Does that also explain why you never saw serialize being called for numeric?

**Comment by rafaelcr** [comment]
Yep, that's right. It was puzzling that even though I specified the `oid` for `numeric` it didn't really call the serializer. I even tried the `oid` for `int8` but no luck.

**Comment by porsager** [comment]
hmm.. I'd like to see that if you can repro..

**Comment by rafaelcr** [comment]
Ok nevermind, so I just did this and the serializer did get called when inserting the number (I think I just did something wrong before):
```typescript
numeric: {
  to: 20, // int8 oid
  from: [20],
  serialize: (x: any) => {
    return x.toString();
  },
  parse: (x: any) => x,
},
```
However, same effect, since of course `toString()` is already the default as you mentioned.

Bottom line is, I think by the point it gets to the serializer it's already too late because postgres has decided the number is a bigint and it doesn't try to set it as numeric. This is why the serializer for oid 1700 didn't work. It will always be tried as an int8 instead of numeric.
